src/lib.rs

```diff
@@ -183,6 +178,46 @@ impl FromIterator<AaConfig> for AaSupport {
     }
 }
 
+/// Errors that can occur in vello.
+#[derive(Error, Debug)]
+pub enum VelloError {
+    /// There is no available device with the features required by Vello.
+    #[cfg(feature = "wgpu")]
+    #[error("Couldn't find suitable device")]
+    NoCompatibleDevice,
+    /// Failed to create surface.
+    /// See [`wgpu::CreateSurfaceError`] for more information.
+    #[cfg(feature = "wgpu")]
+    #[error("Couldn't create wgpu surface")]
+    WgpuCreateSurfaceError(#[from] wgpu::CreateSurfaceError),
+    /// Surface doesn't support the required texture formats.
+    /// Make sure that you have a surface which provides one of
+    /// [`TextureFormat::Rgba8Unorm`] or [`TextureFormat::Bgra8Unorm`] as texture formats.
+    #[cfg(feature = "wgpu")]
+    #[error("Couldn't find `Rgba8Unorm` or `Bgra8Unorm` texture formats for surface")]
+    UnsupportedSurfaceFormat,
+
+    /// Used a buffer inside a recording while it was not available.
+    /// Check if you have created it and not freed before its last usage.
+    #[cfg(feature = "wgpu")]
+    #[error("Buffer '{0}' is not available but used for {1}")]
+    UnavailableBufferUsed(&'static str, &'static str),
+    /// Failed to async map a buffer.
+    /// See [`wgpu::BufferAsyncError`] for more information.
+    #[cfg(feature = "wgpu")]
+    #[error("Failed to async map a buffer")]
+    BufferAsyncError(#[from] wgpu::BufferAsyncError),
+
+    /// Failed to create [`GpuProfiler`].
+    /// See [`wgpu_profiler::CreationError`] for more information.
+    #[cfg(feature = "wgpu-profiler")]
+    #[error("Couldn't create wgpu profiler")]
+    ProfilerCreationError(#[from] wgpu_profiler::CreationError),
+}
+
+#[allow(dead_code)] // this can be unused when wgpu feature is not used
+pub(crate) type Result<T, E = VelloError> = std::result::Result<T, E>;
```

Comment:
This is just a side note but I've always disliked the specialized `Result` types that crates do. It just seems like a bad pattern across the rust ecosystem.

Reply:
I think this is fine here. We're not re-exporting it, which is correct.